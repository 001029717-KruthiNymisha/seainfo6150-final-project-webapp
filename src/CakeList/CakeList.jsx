import React from "react";
import { Link } from "react-router-dom";
import CakeListItem from "../CakeListItem/CakeListItem";
import styles from "./CakeList.module.css";
import PropTypes from "prop-types";

const CakeList = (props) => {
  return (
    <div>
      <div className={styles.pageTitle}>
      <h3>{props.pageTitle}</h3>
      </div>
      <ul className={styles.cakeList}>
        {props.cakes.map((cake,index) => (
          <Link key={index}  to={"/cake/"+cake.Category+"/"+cake.slug}>
            <li key={index}>
              <CakeListItem url={cake.image._url} title={cake.title} />
            </li>
          </Link>
        ))}
      </ul>
    </div>
  );
};
CakeList.propTypes = {
 cakes: PropTypes.array.isRequired
};
export default CakeList;
