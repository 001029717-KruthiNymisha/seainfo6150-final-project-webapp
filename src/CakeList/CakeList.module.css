.cakeList {
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  list-style: none;
  column-gap:20px;
  row-gap: 20px;
  margin-left:-30px;
}
.pageTitle{
font-weight:bold;
font-family:cursive;
text-decoration: underline;
margin-left:80px;
}
@media (min-width: 400px) {
.cakeList{
grid-template-columns:repeat(1, 1fr);
}
}
@media (min-width: 800px) {
.cakeList{
grid-template-columns:repeat(2, 1fr);
}
}
@media (min-width: 1000px) {
.cakeList{
grid-template-columns:repeat(3, 1fr);
}
}