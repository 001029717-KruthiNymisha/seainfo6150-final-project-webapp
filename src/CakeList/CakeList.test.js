import React from "react";
import { render } from "@testing-library/react";
import CakeList from "./CakeList.jsx";
import { BrowserRouter } from 'react-router-dom'

it("renders the CakeList component correctly", () => {
  const test = [
    {
      Category: "GlutenFreeCakes",
      slug: "005-Gluten-Free Happy Birthday Cake",
      title: "Gluten-Free Happy Birthday Cake",
      image: {
        _url: "https://assets.wsimgs.com/wsimgs/rk/images/dp/wcm/202038/0804/img60o.jpg"
      },
    }
  ]

  const { container } = render(
    <BrowserRouter>
      <CakeList
        cakes={Object.values(test)}
        pageTitle="GlutenFreeCakes"
      />
    </BrowserRouter>
   );
    expect(container).toMatchSnapshot();
  });