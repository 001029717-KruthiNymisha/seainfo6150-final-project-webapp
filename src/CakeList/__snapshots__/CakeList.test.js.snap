// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the CakeList component correctly 1`] = `
<div>
  <div>
    <div
      class="pageTitle"
    >
      <h3>
        GlutenFreeCakes
      </h3>
    </div>
    <ul
      class="cakeList"
    >
      <a
        href="/cake/GlutenFreeCakes/005-Gluten-Free Happy Birthday Cake"
      >
        <li>
          <div
            class="cakeListItem"
          >
            <div
              class="cakeImage"
            >
              <img
                alt="Gluten-Free Happy Birthday Cake"
                src="https://assets.wsimgs.com/wsimgs/rk/images/dp/wcm/202038/0804/img60o.jpg"
              />
            </div>
            <div
              class="titleStyle"
            >
              Gluten-Free Happy Birthday Cake
            </div>
          </div>
        </li>
      </a>
    </ul>
  </div>
</div>
`;
